### ISSUE TITLE
Lock screen after long break

### ISSUE THREAD

**Issue by ruXlab** [slgobinath/SafeEyes#102]
I would be great if user can safely leave my computer locked during long break, otherwise user has to wait specified interval or skip break

**Comment by slgobinath** [comment]
Based on my findings, implementing a platform independent lock screen functionality is hard to achieve. There is no unique tool to lock screen across Unity, KDE, Cinnamon etc. ( `gnome-screensaver` didn't work in Manjaroo KDE based on my testing) So I think providing an extension to implement your own lock screen solution is better. Of course, we can provide official extensions to do this later.

**Comment by ruXlab** [comment]
Can it support just some popular screensavers, namely from Unity, KDE, Cinnamon? I might can help to contribute with that.

**Comment by slgobinath** [comment]
You are right @ruXlab.
Earlier I thought to achieve it using extensions for Safe Eyes. But we can add it as part of Safe Eyes. I will start implementing this. You can expect this feature in the next version. I would really appreciate your contribution.
This is the plan:
 - Adding an UI control to enable/disable screen lock feature
 - An additional UI control to define what to do if user Skipped the screen
 - Safe Eyes must sense the desktop environment and execute the relevant command
 - User must have an option to override the default command in case if it does not work

As far as I know, these are some tools:
- Unity: `gnome-screensaver-command -l`
- Cinnamon: `cinnamon-screensaver-command -l`

Need to find as much as possible...
If you want to add more, please comment below.

**Comment by ruXlab** [comment]
From user's point of view I expect check box which disables/enables feature and something like minimum duration of the break which should trigger screen lock. Then it should work seamlessly for all long breaks.
If desktop environment is not supported those controls should disappear. 

What do you think?

**Comment by slgobinath** [comment]
You can expect this feature with your requirements in the next release. (But the next release may take some time as there are some major changes ;-) )

**Comment by slgobinath** [comment]
Hi @ruXlab,

Thanks for you PR. I have made some enhancements to your implementation based on your requirements and my requirements as a user ;-). Now it works like this:
 - Lock screen only after long breaks (Not for short breaks. Because the short breaks are not to leave the computer alone)
 - If this feature is enabled, screen will be locked immediately at the end of break screen or after someone skipped/postponed the break
 - You can skip the break without locking the screen if you skip it within the given time (Default is 5 seconds). In other words, if you skip the break within 5 sec, you will not get the lock screen. So my advice is, leave the computer only after 5 sec (or reduce the time to whatever you like)

What do you think?

**Comment by ruXlab** [comment]
Hey,
> Lock screen only after long breaks (Not for short breaks. Because the short breaks are not to leave the computer alone)

Than it could be just one checkbox "Lock screen for long breaks", without specifying min time before lock. Seems like good simplification and it will work for my use case as well :+1: 

> If this feature is enabled, screen will be locked immediately at the end of break screen or after someone skipped/postponed the break
> You can skip the break without locking the screen if you skip it within the given time (Default is 5 seconds). In other words, if you skip the break within 5 sec, you will not get the lock screen. So my advice is, leave the computer only after 5 sec (or reduce the time to whatever you like)

:+1:  great. Do you want to hard code it or provide another input for user to configure? I believe 5 seconds is okay

**Comment by slgobinath** [comment]
> Do you want to hard code it or provide another input for user to configure? I believe 5 seconds is okay

I have used the spinner defined by you for that purpose. You can give a try by pulling the latest commits. 
One more request, could you please update the Russian translation based on the new change:

- [Lock the screen after every long break](https://github.com/slgobinath/SafeEyes/blob/master/safeeyes/config/lang/ru.json#L29)
- [Maximum time to skip, bypassing the lock screen (in seconds)](https://github.com/slgobinath/SafeEyes/blob/master/safeeyes/config/lang/ru.json#L48)

and there are some more places need to be translated into Russian:

- [Postpone](https://github.com/slgobinath/SafeEyes/blob/master/safeeyes/config/lang/ru.json#L38)
- [Postpone duration (in minutes)](https://github.com/slgobinath/SafeEyes/blob/master/safeeyes/config/lang/ru.json#L39)
- [Safe Eyes desktop file comment](https://github.com/slgobinath/SafeEyes/blob/master/share/applications/safeeyes.desktop)

Since you are satisfied, I am closing this issue but we can continue the chat here if there are anything else related to this issue.

Thanks
